redaxo/src/core/lib/sql/table.php

```diff
@@ -977,4 +977,41 @@ private function resetModified()
             $this->foreignKeysExisting[$foreignKey->getName()] = $foreignKey->getName();
         }
     }
+
+
+
+    /**
+     * Copy the table structure to another table.
+     * @param string $destinationTable
+     *
+     * @throws rex_exception
+     */
+    public function copyTo($destinationTable)
```

Comment:
Ich bin noch nicht so ganz glücklich damit, da es einfach so eine einzelne Methode in der Klasse ist, die aber gar nicht weiter mit der Objektstruktur von `rex_sql_table` arbeitet.

Ich würde mir als Api eher vorstellen (wenn es in diese Klasse integriert wird), dass man erst mal nur eine Kopie des PHP-Objekts erhält, vorbereitet als neue Tabelle, welche man dann mit `create` oder `ensure` erstellt.

Also so in der Art:

```php
$foo = rex_sql_table::get('foo');
$bar = $foo->copyTo('bar');

// ggf noch Änderungen durchführen:
// $bar->ensureColumn(...)

// und erst jetzt wird wirklich was gemacht:
$bar->ensure(); // alternativ: create()
```

So wie die Methode aktuell gestaltet ist, würde sie für mich eher in `rex_sql_util` reinpassen (wo es ja schon `organizePriorities` und so gibt). Was auch ok wäre.

Reply:
@staabm Für was ist dein Daumen? Das bisherige in die `rex_sql_util` verschieben oder das Codebeispiel vom Gregor?

Reply:
ich glaube es wäre am einfachsten wenn wir beide methoden nach rex_sql_util verschieben